akka-cluster/src/main/scala/akka/cluster/Gossip.scala

```diff
@@ -150,11 +152,14 @@ private[cluster] final case class Gossip(
    */
   def merge(that: Gossip): Gossip = {
 
-    // 1. merge vector clocks
-    val mergedVClock = this.version merge that.version
-
-    // 2. merge sets of tombstones
+    // 1. merge sets of tombstones
     val mergedTombstones = tombstones ++ that.tombstones
+    val newTombstonedNodes = mergedTombstones.keySet diff that.tombstones.keySet
+
+    // 2. merge vector clocks (but remove entries for tombstoned nodes)
+    val mergedVClock = newTombstonedNodes.foldLeft(this.version merge that.version) { (vclock, node) ⇒
+      vclock.prune(VectorClock.Node(Gossip.vclockName(node)))
+    }
```

Comment:
This is the thing, we didn't prune versions using the tombstones, so removed nodes vector clocks stayed in the gossip.